### PR TITLE
Remove in_macro_or_desugar

### PR DESCRIPTION
**Pull request by kraai**
`in_macro_or_desugar` is just a wrapper around `Span::from_expansion`, so remove the former and call the latter instead.

changelog: Remove `in_macro_or_desugar`.

### CODE REVIEW

**Comment by kraai** [comment]
I left one occurrence of `in_macro_or_desugar` in `tests/ui/used_underscore_binding.rs` because I wasn't sure whether it should be updated or not.  Please let me know if I should rename that one as well.

File: clippy_lints/src/misc.rs [hunk 1]
```diff
@@ -48,8 +48,8 @@ declare_clippy_lint! {
 declare_clippy_lint! {
     /// **What it does:** Checks for comparisons to NaN.
     ///
-    /// **Why is this bad?** NaN does not compare meaningfully to anything – not
-    /// even itself – so those comparisons are simply wrong.
+    /// **Why is this bad?** NaN does not compare meaningfully to anything   not
+    /// even itself   so those comparisons are simply wrong.
```
Comment (phansch) [inline clippy_lints/src/misc.rs]: It looks like the `-` was replaced with a space here, was this on purpose?

Comment (kraai) [reply clippy_lints/src/misc.rs]: No, that wasn't intentional.  When I save that file with my editor configured to run rustfmt, it replaces the en dashes with spaces.  :(  I've pushed a commit to restore them.

**Comment by phansch** [comment]
Thanks!

@bors r+

File: clippy_lints/src/misc.rs (hunk 1 above)
Comment (kraai) [reply clippy_lints/src/misc.rs]: I was a victim of rust-lang/rust-mode#279.

**Pull request merged**